lib/src/main/java/org/esbtools/lightbluenotificationhook/NotificationHook.java

```diff
@@ -120,69 +120,61 @@ private HookResult processSingleHookDoc(EntityMetadata metadata,
+                                                                      NotificationEntity.ENTITY_VERSION);
+                
+                InsertionRequest newNotification = new InsertionRequest();
+                
+                newNotification.setEntityVersion(notificationVersion);
+                newNotification.setEntityData(objectMapper.valueToTree(notification));
+                
+                LOGGER.debug("Inserting notification");
+                Response response = mediator.insert(newNotification);
+                
+                return new HookResult(notification, response.getErrors(), response.getDataErrors());
+            }
         }
 
         return HookResult.aborted();
     }
 
     /**
-     * Determines if any of the watched fields have changed. If
-     * arrayOrderingSignificant is true, also checks if array
-     * orderings also changed.
+     * Compares the pre- and post- documents after they are passed
+     * through the watchProjector, and returns the delta
      */
-    private boolean watchedFieldsHaveChanged(EntityMetadata metadata,
-                                             JsonDoc preDoc,
-                                             JsonDoc postDoc,
-                                             Projector watchProjector,
-                                             boolean arrayOrderingSignificant) {
+    private DocComparator.Difference<JsonNode> compareDocs(EntityMetadata metadata,
+                                                           JsonDoc preDoc,
+                                                           JsonDoc postDoc,
+                                                           Projector watchProjector) {
         JsonDoc watchedPostDoc = watchProjector.project(postDoc, jsonNodeFactory);
         JsonDoc watchedPreDoc = preDoc == null
             ? new JsonDoc(jsonNodeFactory.objectNode())
             : watchProjector.project(preDoc, jsonNodeFactory);
-
+        
         // Compute diff
         JsonCompare cmp=metadata.getDocComparator();
         try {
-            DocComparator.Difference diff=cmp.compareNodes(watchedPreDoc.getRoot(),watchedPostDoc.getRoot());
+            DocComparator.Difference<JsonNode> diff=cmp.
+                compareNodes(watchedPreDoc.getRoot(),watchedPostDoc.getRoot());
             LOGGER.debug("Diff: {}",diff);
-            if(!diff.same()) {
-                if(diff.getNumChangedFields()>0) {
-                    return true;
-                } else {
-                    // Are there any array element moves?
-                    // Do we care?
-                    if(!diff.getDelta().isEmpty()&&arrayOrderingSignificant) {
-                        return true;
-                    }
-                }
-            }
-            return false;
+            return diff;
         } catch (Exception e) {
             LOGGER.error("Error processing notification",e);
-            return false;
-        }
+            return null;
```

Comment:
I think this can result in NPE. I say let exception propagate and force caller to deal with and return failed HookResult.